### PR TITLE
feat: modify windows AgentBaker E2E pipeline

### PR DESCRIPTION
**Pull request by junjiezhang1997**
**What type of PR is this?**
/kind improvement

**What this PR does / why we need it**:
1. Exit directly when e2e-starter.sh fails.
2. Check whether linux file already exists when cluster is already created.
3. Fix bug introduced by the change of csepackage folder name.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: e2e/windows/e2e-helper.sh [hunk 1]
```diff
@@ -81,10 +81,26 @@ upload_linux_file_to_storage_account() {
...
 
 download_linux_file_from_storage_account() {
+    if [[ "$(check_linux_file_exists_in_storage_account)" == *"Linux file already exists in storage account."* ]]; then
+        linuxFileURL="https://${AZURE_E2E_STORAGE_ACCOUNT_NAME}.blob.core.windows.net/${WINDOWS_E2E_STORAGE_CONTAINER}/${MC_VMSS_NAME}-linux-file.zip"
+
+        export AZCOPY_AUTO_LOGIN_TYPE="MSI"
+        export AZCOPY_MSI_RESOURCE_STRING="${AZURE_MSI_RESOURCE_STRING}"
+
+        array=(azcopy_*)
+        ${array[0]}/azcopy copy $linuxFileURL file.zip
```
Comment (AbelHu) [inline e2e/windows/e2e-helper.sh:96]: can we use `az storage blob download --auth-mode login`?

Comment (junjiezhang1997) [reply e2e/windows/e2e-helper.sh:96]: Sure. I will update it.

File: e2e/windows/e2e-helper.sh [hunk 2]
```diff
@@ -81,10 +81,26 @@ upload_linux_file_to_storage_account() {
...
     else
         err "Failed to upload linux file. Error code is $retval."
+        exit 1
     fi
 }
 
 download_linux_file_from_storage_account() {
+    local retval
+    retval=0
+    if [[ "$(check_linux_file_exists_in_storage_account)" == *"Linux file already exists in storage account."* ]]; then
```
Comment (AbelHu) [inline e2e/windows/e2e-helper.sh:91]: Suggest using `isRemoteCsePackageExist=$(az storage blob exists --auth-mode login --blob-url $cseBlobUrlForUploading | jq -r '.exists')`

We have the code in e2e/windows/e2e-scenario.sh. You can wrap it as a function

Comment (junjiezhang1997) [reply e2e/windows/e2e-helper.sh:91]: check_linux_file_exists_in_storage_account is just a function. I have changed the way of checking file existence in this function.

**Review by AbelHu — APPROVED** [review]
LGTM. Thanks for the PR

**Pull request merged**